src/utils/src/ExponentialBackoffRetryStrategy.c

```diff
@@ -0,0 +1,245 @@
+#include "Include_i.h"
+
+STATUS initializeDefaultExponentialBackoffConfig(PExponentialBackoffRetryStrategyConfig pExponentialBackoffRetryStrategyConfig) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(pExponentialBackoffRetryStrategyConfig != NULL, STATUS_NULL_ARG);
+
+    pExponentialBackoffRetryStrategyConfig->maxRetryCount = KVS_INFINITE_EXPONENTIAL_RETRIES;
+    pExponentialBackoffRetryStrategyConfig->maxRetryWaitTime = HUNDREDS_OF_NANOS_IN_A_MILLISECOND * DEFAULT_KVS_MAX_WAIT_TIME_MILLISECONDS;
+    pExponentialBackoffRetryStrategyConfig->retryFactorTime = HUNDREDS_OF_NANOS_IN_A_MILLISECOND * DEFAULT_KVS_RETRY_TIME_FACTOR_MILLISECONDS;
+    pExponentialBackoffRetryStrategyConfig->minTimeToResetRetryState = HUNDREDS_OF_NANOS_IN_A_MILLISECOND * DEFAULT_KVS_MIN_TIME_TO_RESET_RETRY_STATE_MILLISECONDS;
+    pExponentialBackoffRetryStrategyConfig->jitterFactor = DEFAULT_KVS_JITTER_FACTOR_MILLISECONDS;
+
+CleanUp:
+    LEAVES();
+    return retStatus;
+}
+
+STATUS resetExponentialBackoffRetryState(PExponentialBackoffRetryStrategyState pExponentialBackoffRetryStrategyState) {
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+
+    CHK(pExponentialBackoffRetryStrategyState != NULL, STATUS_NULL_ARG);
+
+    pExponentialBackoffRetryStrategyState->currentRetryCount = 0;
```

Comment:
Can this API and `exponentialBackoffRetryStrategyBlockingWait` be called by two separate threads?  If so `lastRetryWaitTime` and `lastRetrySystemTime` are UINT64 which means it is not an atomic operation to set a value (which this API and `exponentialBackoffRetryStrategyBlockingWait` both do, on 32-bit systems. This will result in a torn right.  So we need to be absolutely sure this cannot happen in 2 different threads OR you need to introduce a mutex to protect the setting of these values.